kroxylicious/src/main/java/io/kroxylicious/proxy/config/PluginFactory.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.proxy.config;
+
+import io.kroxylicious.proxy.plugin.UnknownPluginInstanceException;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * A PluginFactory is able to resolve reference to a plugin implementation (ie. a name) to that implementation.
```

Comment:
This sounds like the description of a plug-in resolver rather than something that creates instances.